framework/src/base/MooseVariableBase.C

```diff
@@ -46,9 +46,24 @@ MooseVariableBase::name() const
   return _sys.system().variable(_var_num).name();
 }
 
+std::vector<dof_id_type>
+MooseVariableBase::allDofIndices() const
```

Comment:
It's fine to have it here... it should just use the threaded loop object and do parallel communication